python/adelphi/adelphi/nb.py

```diff
@@ -89,7 +91,12 @@ def __init__(self, cluster, props):
 
         self.rampup_cycles = props["rampup-cycles"]
         self.main_cycles = props["main-cycles"]
-        self.numeric_max = min((self.rampup_cycles + self.main_cycles) * 1000, MAX_NUMERIC_VAL)
+        self.scenario_name = props["scenario-name"]
+        self.numeric_max = \
+            min((self.rampup_cycles + self.main_cycles) * 1000, MAX_NUMERIC_VAL) \
+            if self.rampup_cycles and self.main_cycles \
+            else MAX_NUMERIC_VAL
```

Comment:
This reflects the maximum size we can get for numeric values.  If the number of cycles is specified we can use this to constrain the number of values in order to increase the likelihood of hits when we do selects on primary keys.